mag3110/mag3110.go

```diff
@@ -49,8 +49,8 @@ func (d Device) ReadMagnetic() (x int16, y int16, z int16) {
 }
 
 // ReadTemperature reads the current die temperature in degrees Celsius.
-func (d Device) ReadTemperature() (temp int8) {
+func (d Device) ReadTemperature() (int32, error) {
 	data := make([]byte, 1)
 	d.bus.ReadRegister(uint8(d.Address), DIE_TEMP, data)
-	return int8(data[0])
+	return int32(data[0]), nil
```

Comment:
This is in units of 1°C, not in units of 0.001°C as described in https://github.com/tinygo-org/drivers/wiki/Driver-design---API. While the signature may be the same, the interface works differently.

Other than that, this is a good change.

Reply:
OK updated so the functionality matches by returning the correct scaling of the result.